include/dev/HUDL.hpp

```diff
@@ -137,194 +152,82 @@ class HUDL {
         /**
-                 * RampUP Board RPODO 0 Mapping
-                 * Determines the PDO messages to receive when RPDO0 is triggered
-                 * 0: The number of PDO message associated with the RPDO
-                 * 1: Voltage One
-                 * 2: Voltage Two
-                 * 3: Voltage Three
-                 * 4: Voltage Four
-                 */
-        {
-            .Key = CO_KEY(0x1600, 0, CO_UNSIGNED8 | CO_OBJ_D__R_),
-            .Type = nullptr,
-            .Data = (uintptr_t) 4,
-        },
-        {
-            .Key = CO_KEY(0x1600, 1, CO_UNSIGNED32 | CO_OBJ_D__R_),
-            .Type = nullptr,
-            .Data = CO_LINK(0x2100, 0, 16),// Voltage One
-        },
-        {
-            .Key = CO_KEY(0x1600, 2, CO_UNSIGNED32 | CO_OBJ_D__R_),
-            .Type = nullptr,
-            .Data = CO_LINK(0x2100, 1, 16),// Voltage Two
-        },
-        {
-            .Key = CO_KEY(0x1600, 3, CO_UNSIGNED32 | CO_OBJ_D__R_),
-            .Type = nullptr,
-            .Data = CO_LINK(0x2100, 2, 16),// Voltage Three
-        },
-        {
-            .Key = CO_KEY(0x1600, 4, CO_UNSIGNED32 | CO_OBJ_D__R_),
-            .Type = nullptr,
-            .Data = CO_LINK(0x2100, 3, 16),// Voltage Four
-        },
+         * RampUP Board RPDO 0 Mapping
+         * Determines the PDO messages to receive when RPDO0 is triggered
+         * 0: The number of PDO message associated with the RPDO
+         * 1: Voltage One
+         * 2: Voltage Two
+         * 3: Voltage Three
+         * 4: Voltage Four
+         */
+        RECEIVE_PDO_MAPPING_START_KEY_16XX(0, 4),
+        RECEIVE_PDO_MAPPING_ENTRY_16XX(0, 1, PDO_MAPPING_UNSIGNED16),
+        RECEIVE_PDO_MAPPING_ENTRY_16XX(0, 2, PDO_MAPPING_UNSIGNED16),
+        RECEIVE_PDO_MAPPING_ENTRY_16XX(0, 3, PDO_MAPPING_UNSIGNED16),
+        RECEIVE_PDO_MAPPING_ENTRY_16XX(0, 4, PDO_MAPPING_UNSIGNED16),
+
         /**
-                 * RampUP Board RPDO 1 Mapping
-                 * Determines the PDO messages to receive when RPDO1 is triggered
-                 * 0: The number of PDO message associated with the RPDO
-                 * 1: Temperature
-                 */
-        {
-            .Key = CO_KEY(0x1601, 0, CO_UNSIGNED8 | CO_OBJ_D__R_),
-            .Type = nullptr,
-            .Data = (uintptr_t) 1,
-        },
-        {
-            .Key = CO_KEY(0x1601, 1, CO_UNSIGNED32 | CO_OBJ_D__R_),
-            .Type = nullptr,
-            .Data = CO_LINK(0x2101, 0, 16),// Temperature
-        },
+         * RampUP Board RPDO 1 Mapping
+         * Determines the PDO messages to receive when RPDO1 is triggered
+         * 0: The number of PDO message associated with the RPDO
+         * 1: Temperature
+         */
+        RECEIVE_PDO_MAPPING_START_KEY_16XX(1, 1),
+        RECEIVE_PDO_MAPPING_ENTRY_16XX(1, 1, PDO_MAPPING_UNSIGNED16),
+
         /**
-                 * *** START VARIABLE LINKING ***
-                 */
+         * *** START VARIABLE LINKING ***
```

Comment:
```suggestion
         * *** START DATA LINKING ***
```
This is just to use the same verbiage as the CANopen stack